py_mypy/README.md

```diff
@@ -0,0 +1,109 @@
+# Using MyPy with rules_python
+
+This is a walkthrough of using [rules_mypy](https://github.com/theoremlp/rules_mypy) together with `rules_python` to apply typechecks as part of "building" a Python application.
+
+## How MyPy will work
+
+Bazel's [aspects](https://bazel.build/extending/aspects) allow extensions to traverse the build graph and apply rewrites to it between the analysis pass and before `build` happens.
+A common application of aspects is to "bolt on" behavior to existing rules without having to modify them.
+Which is exactly what we're going to do here.
+
+One way that an aspect can extend an existing rule is by adding an [`OutputGroupInfo`](https://bazel.build/versions/7.4.0/rules/lib/providers/OutputGroupInfo) provider to the rule.
+Output groups are a slightly unusual feature which allows for rule names to be overloaded, and for a rule to provide multiple kinds of outputs.
+To take a slightly familiar example, the `py_binary` rule normally outputs a launcher script and a `.runfiles` tree, but it also provides a zipapp output which can be selectively enabled.
+Outputs may be selected during a build using the [`--output_groups`](https://bazel.build/reference/command-line-reference#flag--output_groups) flag.
```

Comment:
Add "and can be referenced by other targets by using a `filegroup` with the `output_group` attribute"